### PR TITLE
Adding warning before disabling the guest accounts access

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Adding warning before disabling the guest accounts access

#### Ticket Link
[MM-18443](https://mattermost.atlassian.net/browse/MM-18443)

#### Related Pull Requests
- Has server changes (not-added-yet)

#### Screenshots
image

### CODE REVIEW

**Comment by srkgupta** [comment]
Hi @jespino 
On the PR instance, I am not able to Enable the Guest Access feature. If I try to enable the setting and save and then reload, the Guest Access feature shows as disabled. Can you please check and fix this.

**Comment by jespino** [comment]
@srkgupta I think the problem was because the fixes for that were in a commit that had other problems (linting and i18n) do you mind to check again in the updated spinwick?

**Comment by srkgupta** [comment]
Thanks @jespino. It works fine now on the latest PR instance. Will validate the rest of the ticket now.

File: components/admin_console/custom_enable_disable_guest_accounts_setting.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+import ConfirmModal from 'components/confirm_modal.jsx';
+
+import BooleanSetting from './boolean_setting';
+
+type Props = {
+    id: string;
+    value: boolean | string;
```
Comment (streamer45) [inline components/admin_console/custom_enable_disable_guest_accounts_setting.tsx]: Not strictly related to this PR but why do we have to handle string booleans?

File: components/admin_console/custom_enable_disable_guest_accounts_setting.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                defaultMessage='When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests.'
+            />
+        );
+
+        let value;
+        if (typeof this.props.value === 'string') {
+            value = this.props.value === 'true';
+        } else {
+            value = this.props.value;
+        }
```
Comment (streamer45) [inline components/admin_console/custom_enable_disable_guest_accounts_setting.tsx]: In cases like this I am a fan of the ternary operator:
```js
const value = (this.props.value === 'true' || this.props.value === true) ? true : false;
```

File: components/admin_console/custom_enable_disable_guest_accounts_setting.tsx (hunk 1 above)
Comment (jespino) [reply components/admin_console/custom_enable_disable_guest_accounts_setting.tsx]: Done, not needed to have both.

File: components/admin_console/custom_enable_disable_guest_accounts_setting.tsx (hunk 2 above)
Comment (jespino) [reply components/admin_console/custom_enable_disable_guest_accounts_setting.tsx]: This is no longer needed

File: i18n/en.json [hunk 3]
```diff
@@ -681,6 +681,9 @@
   "admin.group_settings.ldapGroupsTitle": "AD/LDAP Groups",
   "admin.guest_access.enableDescription": "When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests.",
   "admin.guest_access.enableTitle": "Enable Guest Access: ",
+  "admin.guest_access.disableConfirmButton": "Disable Guest Access",
+  "admin.guest_access.disableConfirmMessage": "Disabling guest access will revoke all current Guest Account sessions. Guests will no longer be able login and new guests cannot be invited into Mattermost. Guest users will be marked as inactive in user lists. Enabling this feature will not reinstate previous guest accounts.",
```
[the remark this answers is not in the document]
Comment (jespino) [reply i18n/en.json]: Done

**Pull request merged**